pkg/agent/agent.go

```diff
@@ -319,7 +319,12 @@ func getAgentResource() (*apiV1.ResourceInstance, error) {
 
 	response, err := agent.apicClient.ExecuteAPI(coreapi.GET, agentResourceURL, nil, nil)
 	if err != nil {
-		return nil, err
+		// if agentResourceType is already a standard generic agent resource
+		if agentResourceType == agentTypesMap[agent.cfg.AgentType] {
+			return nil, err
+		}
+		AgentResourceType = ""
+		return getAgentResource()
```

Comment:
Isn't this calling itself?

Reply:
yes, but resetting the resource type so we look for the generic type rather than dataplane specific type

Reply:
Probably deserves a big, fat comment in the code then. It certainly isn't self-documenting as it is.

Reply:
Agreed